modules/dynamic/test/src-jvm/smithy4s/dynamic/model/JsonIOProtocol.scala

```diff
@@ -0,0 +1,83 @@
+package smithy4s
+package dynamic.model
+
+import cats.effect.IO
+import cats.syntax.all._
+
+/**
+  * These are toy interpreters that turn services into json-in/json-out
+  * functions, and vice versa.
+  *
+  * Created for testing purposes.
+  */
+object JsonIOProtocol {
```

Comment:
@kubukoz, I've created this to facilitate testing. Given a service given by the dynamic model, we can create a "proxy" function, that round trips a `Document => IO[Document]` kleisli through the `fromJsonIO` and `toJsonIO` functions. 

Then, if we take that proxy and create a corresponding statically generated stub from it, in theory the data should go through the proxy back and forth, in an untampered fashion. 

We can add some middleware to intercept and store the incoming and outgoing json in a CE-Ref, and either perform assertions on the json itself, or use it for logging purposes when we see that the data has not gone through the dynamic proxy as expected.